python/cudf/cudf/core/series.py

```diff
@@ -529,22 +536,39 @@ def append(self, other, ignore_index=False):
         -------
         A new Series equivalent to self concatenated with other
         """
-        this = self
-        other = Series(other)
+        if verify_integrity not in (None, False):
+            raise NotImplementedError(
+                "verify_integrity parameter is not supported yet."
+            )
 
-        from cudf.core.column import numerical
-        from cudf.utils.dtypes import numeric_normalize_types
+        if isinstance(to_append, (list, tuple)):
+            to_concat = [self]
+            to_concat.extend(to_append)
+        else:
+            to_concat = [self, to_append]
+
+        dtype_mismatch = False
+        for obj in to_concat:
+            if not dtype_mismatch and self.dtype != obj.dtype:
+                dtype_mismatch = True
```

Comment:
Need to ensure that everything in `to_concat` is a Series / Index / etc. before checking the `.dtype` attribute.

Reply:
This will now go through `cudf.concat` and have `Series` validation and the `dtype` check will happen in `Series._concat`.